src/main/java/io/camunda/testing/assertions/ProcessInstanceAssertions.java

```diff
@@ -388,4 +391,34 @@ public ProcessInstanceAssertions isNotWaitingForMessage(final String... messageN
             });
     return openMessageSubscriptions;
   }
+
+  public ProcessInstanceAssertions hasVariable(final String name) {
+    extractVariables()
+        .withFailMessage(
+            "Process with key %s does not contain variable with name %s",
+            actual.getProcessInstanceKey(), name)
+        .containsKey(name);
+    return this;
+  }
+
+  public ProcessInstanceAssertions hasVariableWithValue(final String name, final String value) {
```

Comment:
:x: the second parameter needs to be on Object. We need to convert the object value to a JSON string (ideally in exactly the same way that Zeebe would) and then compare the two strings

Reply:
You can try an example where the type is a String. Then you need to assert `hasVariableWithValue("test", "\"value\"")` which is not intuitive.

Reply:
`ZeebeObjectMapper`will do most of the work